eck.php

```diff
@@ -26,6 +26,17 @@ function eck_civicrm_config(&$config) {
   _eck_civix_civicrm_config($config);
 }
 
+/**
+ * Convert ECK EntityType name to sql table name.
+ *
+ * @param string $entityTypeName
+ * @return string
+ */
+function _eck_get_table_name(string $entityTypeName): string {
+  // SQL table names must be alphanumeric and no longer than 64 characters
+  return CRM_Utils_String::munge('civicrm_eck_' . strtolower($entityTypeName), '_', 64);
+}
```

Comment:
We're currently not validating name collisions, do we; shall we?

Reply:
Yes we do. There is a unique index on `EntityType.name`:

https://github.com/systopia/de.systopia.eck/blob/a6cb9b58d59ff12a644d928477da8f2524fa540e/xml/schema/CRM/Eck/EckEntityType.xml#L36-L40

Reply:
@jensschuppe `EntityType.name` has enforced uniqueness at the DB level, however it has a maxlength of 255 which then would get trimmed by this function to `64 - strlen('civicrm_eck_')` which is 58 characters. So in theory any entities beginning with the same 58 characters could cause name collisions with identical db table names.
I think we ought to reduce the length of the `name` column to 58 characters. I can do that in a followup PR after you merge this.